src/modules/edc-demo/components/contract-viewer/contract-viewer.component.ts

```diff
@@ -68,28 +64,16 @@ export class ContractViewerComponent implements OnInit {
   }
 
   onTransferClicked(contract: ContractAgreementDto) {
-    const dialogRef = this.dialog.open(CatalogBrowserTransferDialog);
+    const data: ContractAgreementTransferDialogData = {
+      contractId: contract.id,
+    };
+    const dialogRef = this.dialog.open(ContractAgreementTransferDialog, {data});
 
     dialogRef
       .afterClosed()
-      .pipe(
-        map((result) => result.dataDestination as DataAddressDto),
-        first(),
-        switchMap((dataAddressDto) =>
-          this.contractAgreementService.initiateTransfer(
-            contract.id,
-            dataAddressDto,
-          ),
-        ),
-      )
-      .subscribe({
-        next: (transferId) => {
-          this.startPolling(transferId, contract.id!);
-        },
-        error: (error) => {
-          console.error(error);
-          this.notificationService.showError('Error initiating transfer');
-        },
+      .pipe(filter((it) => !!it))
+      .subscribe((result: ContractAgreementTransferDialogResult) => {
+        this.startPolling(result.transferProcessId, result.contractId);
       });
```

Comment:
No error message, when the transfer fails?

Reply:
This code block is only reached after success. The dialog handles errors / success and only closes itself with the success result message when it did successfully initiate the transfer. This is so that the dialog doesn't have to close to submit its data.